pkg/model/crd.go

```diff
@@ -236,6 +241,17 @@ func (r *CRD) SpecFieldNames() []string {
 	return res
 }
 
+// SpecFieldShapeNames returns a sorted slice of shape names for each of the
+// Spec fields
+func (r *CRD) SpecFieldShapeNames() []string {
+	res := make([]string, 0, len(r.SpecFields))
+	for _, field := range r.SpecFields {
+		res = append(res, field.ShapeRef.ShapeName)
+	}
+	sort.Strings(res)
+	return res
+}
+
```

Comment:
This isn't needed, AFAICT, nor is the `CRD.IsShapeInSpec` method below since the block of code you added above in `CRD.HasShapeAsMember` effectively will return true when the CRD has a shape as a member (or nested member). I believe you only added this as a caching mechanism in the GetTypeDefs() function, is that right?

I'd prefer if we kept the code as simple as possible and so think you can remove this and the `IsShapeInSpec` methods.

Reply:
Oh wow. You are absolutely correct. My change to `CRD.HasShapeAsMember` came after I added these other methods, and that is the only thing that was necessary. Thank you!